rs/nns/governance/tests/proposals.rs

```diff
@@ -0,0 +1,206 @@
+                proposal_data.id = Some(ProposalId { id: 79 });
+
+                let proposal = proposal_data.proposal.as_mut().unwrap();
+                proposal.action = Some(Action::Motion(Default::default()));
+                proposal_data
+            },
+        };
+
+        // Verify all ReadyToSettle.
+        assert_eq!(
+            proposals
+                .values()
+                .map(|proposal| proposal.reward_status(NOW_SECONDS, 4 * ONE_DAY_SECONDS))
+                .collect::<Vec<_>>(),
+            // Using all is more concise, but if the assert fails, assert! gives
+            // less diagnostic information than assert_eq!.
+            vec![
+                ProposalRewardStatus::ReadyToSettle,
+                ProposalRewardStatus::ReadyToSettle,
+                ProposalRewardStatus::ReadyToSettle,
+            ],
+        );
+
+        // Verify reward weights.
+        assert_eq!(
+            proposals
+                .values()
+                .map(|proposal| proposal.topic().reward_weight())
+                .collect::<Vec<f64>>(),
+            [1.0, 1.0, 20.0]
+        );
+
+        proposals
+    };
+
+    static ref GOVERNANCE_PROTO: GovernanceProto = GovernanceProto {
+        neurons: NEURONS.clone(),
+        proposals: PROPOSALS.clone(),
+        genesis_timestamp_seconds: NOW_SECONDS - REWARD_DISTRIBUTION_PERIOD_SECONDS,
+        economics: Some(NetworkEconomics::with_default_values()),
+        ..Default::default()
+    };
+}
+
+#[tokio::test]
+async fn test_distribute_rewards_with_total_potential_voting_power() {
+    // Step 1: Prepare the world.
+
+    let fake_driver = fake::FakeDriver::default()
+        .at(NOW_SECONDS)
+        // This ensures that the rewards purse is exactly equal to 2x the total
+        // weighted voting power. That way, the amount of rewards that each
+        // neuron gets is equal to its voting power (times the reward weight of
+        // the proposal).
+        .with_supply(Tokens::from_e8s(5844002454));
+
+    let mut governance = Governance::new(
+        GOVERNANCE_PROTO.clone(),
+        fake_driver.get_fake_env(),
+        fake_driver.get_fake_ledger(),
+        fake_driver.get_fake_cmc(),
+    );
+
+    // Step 2: Call code under test.
+    governance.run_periodic_tasks().await;
+
+    // Step 3: Inspect result(s).
+    let get_neuron_rewards = |neuron_id| {
+        governance
+            .with_neuron(&NeuronId { id: neuron_id }, |neuron| {
+                neuron.maturity_e8s_equivalent
+            })
+            .unwrap()
+    };
+
+    assert_eq!(get_neuron_rewards(1042), 0); // Didn't vote -> no rewards.
+
+    let reward_ratio = get_neuron_rewards(1043) as f64 / get_neuron_rewards(1044) as f64;
+
+    let weighted_voting_power_ratio = (2 * 2_000 + 20 * 2_000) as f64 /   // neuron 1043
+        (2 * 30_000 + 20 * 30_000) as f64; // neuron 1044
+
+    let error = (reward_ratio - weighted_voting_power_ratio) / weighted_voting_power_ratio;
```

Comment:
I don't see how this tests the new functionality... it just tests the proportion of rewards between the neurons, which would be the same whether or not the total_potential_voting_power was added, correct?

I feel like I may be missing something.